### PR TITLE
Rename repo from data-viewer to front-end

### PR DESCRIPTION
**Pull request by jasonaowen**
Update all the places that the repository name occurs. This does not update any in-app user-facing strings, per @kfogel's [suggestion](https://github.com/PhilanthropyDataCommons/data-viewer/issues/315#issuecomment-1716320859).

Issue #315 Choose new name for Data Viewer

### CODE REVIEW

**Comment by jasonaowen** [comment]
We'll also need to rename the repo in the GitHub admin view; that can happen before or after this is merged, but should be close in time so that the "Recent Changes" URL continues to work.

File: README.md
Comment (slifty) [inline README.md:1]: This was auto-populated by github and while we're here changing it it might be nice to rename the title to something more human like "Philanthropy Data Commons Front End" (the service is currently titled "Philanthropy Data Commons Service")

**Review by slifty — CHANGES_REQUESTED** [review]
I found a few more:

* https://github.com/PhilanthropyDataCommons/data-viewer/blob/2455150ec98224b655d95db4591d95edc71ddce7/public/index.html#L15
* https://github.com/PhilanthropyDataCommons/data-viewer/blob/2455150ec98224b655d95db4591d95edc71ddce7/public/manifest.json#L3
* https://github.com/PhilanthropyDataCommons/data-viewer/blob/2455150ec98224b655d95db4591d95edc71ddce7/src/pages/Landing.tsx#L23

Not sure if we care about this one:

* https://github.com/PhilanthropyDataCommons/data-viewer/blob/2455150ec98224b655d95db4591d95edc71ddce7/src/interfaces/DataViewerProposal.ts#L1

Also, a suggestion to give the ReadMe a more robust title rather than using the repo slug.

**Comment by slifty** [comment]
Seeing now that you intentionally skipped those -- I thought Karl was suggesting we not get bogged down thinking about branding, but I didn't think he meant we should keep the name Data Viewer for the frontend.

What about renaming it to PDC Data Portal for user facing strings (or just calling it the Philanthropy Data Commons) as part of this PR?  That doesn't prevent us from being able to change it again later, and it would still accomplish karl's suggestion of decoupling the public name from the repository name.

**Comment by reefdog** [comment]
@slifty I think it would be trivial to do any suggested branding/user-facing changes as a follow-up PR. Jason's changes are essentially non-controversial ("change IDs and repo references"), while your suggested changes warrant [even a tiny amount of] discussion. For that reason alone I suggest merging this one in and then following up right away with your suggestion!

**Review by reefdog — APPROVED** [review]
While acknowledging @slifty's desire to update user-facing names / branding, I do think this is a good PR as-is and am signaling _my_ vote by approving it. (That said, Dan if you feel stronger then by all means you can still formally request changes!)

File: README.md
Comment (reefdog) [reply README.md:1]: I do think (whether in this PR or a followup) that re-titling the README is a good idea.

**Comment by reefdog** [comment]
@slifty:

> Not sure if we care about [`DataViewerProposal`]

Ooh, good catch. I actually *do* think we care about it and it should be renamed `FrontEndProposal`, but also, it's a small red flag that we have to rename an interface just because the project name changed. Though it does _kinda_ make sense, because this is the interface for "how this front end models a proposal" (as opposed to how the back-end service models it).

File: README.md
Comment (jasonaowen) [reply README.md:1]: Let's do that as part of the follow-up rebrand PR!

**Comment by slifty** [comment]
For sure -- my `request changes` came before realizing it was intended - happy to mark as approved (though yeah the DataViewerProposal update makes sense to make if others agree!)

File: README.md
Comment (slifty) [reply README.md:1]: FWIW I don't think the markdown title would be part of the rebrand since it is a dev document.  If there's a reason we don't want to change this to "Philanthropy Data Commons Front End" now that's fine, but I don't think this file should ever say "Philanthropy Data Commons Data Portal" or whatever we end up landing on for the branded name.

**Review by slifty — APPROVED** [review]
I left my feedback!  Take it or leave it but I will not stand in the way of progress 😅

File: README.md
Comment (reefdog) [reply README.md:1]: I do agree this file's title will almost certainly differ from the landing page title, but I would reframe the follow-up PR from "branding" (even though it involves that) to "user-facing language", where users are both website visitors and devs reading documentation. In that case, both the language on the landing page and the README title are fair game (IMOoooocow).

Comment (reefdog) [reply README.md:1]: Whoops I edited out my soft-landing "What do you think?" when rearranging sentences, so…

What do you think?

Comment (slifty) [reply README.md:1]: > I feel like the common convention is to have the title match the repo name

Got it!  I had always interpreted that as more of "it's the default and nobody got around to making a real title" BUT I'm also fine just punting / leaving as proposed!

Carry forth!  The bike shed is beautiful regardless!

Comment (jasonaowen) [reply README.md:1]: :bike: :house: 

I did a little further research and didn't find anything particularly persuasive in one direction or the other! So, now that I have your blessing, I'm going to pull the rip cord on this and leave it alone for now. :blue_heart: :parachute:

File: src/pages/Landing.tsx
Comment (jasonaowen) [inline src/pages/Landing.tsx:76]: I just tested this by clicking through the deploy preview and the new URL is now live.

Comment (reefdog) [reply src/pages/Landing.tsx:76]: Sigh how I wish we could wildcard the auth domain. :/

Comment (jasonaowen) [reply src/pages/Landing.tsx:76]: Me, too. Someday...

Actually, this might be worth re-evaluating separately - it might be worthwhile to do some legwork to enable deploy preview logins automatedly! It won't be out-of-the-box, but I think it should be possible.

**Review by reefdog — APPROVED** [review]
The code scans cleanly and I pulled locally / logged in to be sure the interface change still worked. It does indeed!

File: README.md
Comment (kfogel) [reply README.md:1]: > Carry forth! The bike shed is beautiful regardless!

Obligatory https://fuchsia.bikeshed.com/ .

**Pull request merged**
